tests/test_annotations.py

```diff
@@ -584,6 +585,11 @@ class A:
         assert typing.List[int] == attr.fields(A).b.type
         assert typing.List[int] == attr.fields(A).c.type
 
+    @pytest.mark.skipif(
+        PY_3_14_PLUS,
+        reason="Forward references are changing a lot in 3.14. "
+        "Passes only for slots=True",
```

Comment:
Why not call `pytest.xfail()` conditionally, then?

Reply:
because it passes with slots=True – can I refer to other fixtures in the xfail condition?

Reply:
The `pytest.xfail()` helper is not a marker (`pytest.mark.xfail()`). You have to call it inside the test function body and it's more flexible since conditionals are in the code.

Reply:
You can use xfail in function body like this:
```python
if PY_3_14_PLUS and slots is not True:
    pytest.xfail("")
```
(But I doubt it's doable with a decorator.)

Reply:
True, but that's not a real xfail tho. I like strict xfails that start failing once the test isn't failing anymore, but sure, it's semantically more correct.

Reply:
I suppose what you want could be implemented with indirect fixtures or other hooks. That would be a bit more complicated, though.